core/DataTable/Filter/ColumnDelete.php

```diff
@@ -161,4 +145,46 @@ public function filter($table)
 
         return $table;
     }
+
+    /**
+     * @param $table
+     * @return array
+     */
+    protected function removeColumnsFromTable(&$table)
+    {
+        if(!$this->isArrayAccess($table)) {
+            return;
+        }
+        foreach ($table as $index => &$row) {
+            if(!$this->isArrayAccess($row)) {
```

Comment:
I presume you are doing this since it can now be a scalar etc. I'd definitely recommend testing performance for this. I remember from several performance profiles that showColumns/deleteColumns was always slow and one of the major contributors (like top 3 contributors) to slow performance in Piwik and now iterating over all array keys etc may make it much slower and we may want to find a different way for arrays vs dataTables etc.